src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEFAULT_SIZE];
+        Arrays.fill(stack, null);
```

Comment:
There is no need to fill the stack array with nulls during initialization. New arrays in Java are already filled with nulls.